App/Redux/PreferencesRedux.ts

```diff
@@ -125,6 +151,12 @@ export function reducer (state: PreferencesState = initialState, action: Prefere
       service.status = action.payload.status === undefined ? !service.status : action.payload.status
       return {...state, services: {...state.services, [action.payload.name]: service}}
     }
+    case getType(actions.toggleStorageRequest): {
+      const storageType = state.storage[action.payload.name]
+      if (!storageType) { return state }
+      storageType.status = action.payload.status === undefined ? !storageType.status : action.payload.status
+      return {...state, storage: {...state.storage, [action.payload.name]: storageType}}
```

Comment:
Todo: we can set the first local photo date the moment that the option is toggled. in that way we'll never get photos older than the option change. you can also toggle it off and then back on again and we'll skip all that happened in between

Reply:
ah, this is in there now actually. no longer todo